types/react-router-navigation-core/tslint.json

```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "interface-over-type-literal": false,
```

Comment:
It would be better to just use the default tslint rules instead of adding exceptions for some packages.

Reply:
this also has to do with my preference of types over interfaces, but if there are good arguments for the different rules I can get convinced 😄

Reply:
Intersecting two types never fails, while interfaces bring more guarantees.
Also, when you use an interface we are able to use the type name in more places.
```ts
type T = { a: number; x: number };
type U = T & { b: number; x: string };
declare const u: U;
u.a; // (property) a: number
u.b; // (property) b: number
u.x; // (property) x: number & string

interface I { a: number; x: number; }
interface J extends I { b: number; x: string; } // Error, bad override of 'x'
declare const j: J;
j.a; // (property) I.a: number
j.b; // (property) J.b: number
j.x; // (property) J.x: string
```

Reply:
okay I see the benefit with the bad override, thank you for the explanation. But still don't know if I'm completely convinced, because interfaces allow declaration merging what I think is quite dirty, because it spreads the definition to different places.
Also the behavior of intersection types is very close to the behavior of object merging in JS, so I think of it as the typed equivalent of type T3 = T1 & T2 Object.assign(T1, T2)

In this specific case I still want to stay with types instead of interfaces just to stay closer to the original code in flowType. 
https://github.com/LeoLeBras/react-router-navigation/blob/master/packages/react-router-navigation/src/TypeDefinitions.js 

One benefit of types is that the behavior is closer to the runtime behavior, object merging is also quite radical and the last property wins 😉

Reply:
Unless the value is actually `number & string` at runtime, it's not more accurate to the runtime behavior! The compiler and language service work best on named over anonymous types -- please stick to the DefinitelyTyped lint rules.

Reply:
hmm you are making it more difficult than it should be to contribute type declarations. 
As I already said the main reason to use types instead of interfaces is that the original code written in FlowType also uses types instead of interfaces.

If I would use interfaces everytime the original code changes there will be the transfer from type to interface again. By using types it is almost just copy and paste.

Reply:
another argument to stay with types for this lib is, that the implementation of intersection types in flow is the same as in typescript, with the same flaws (https://flow.org/en/docs/types/intersections/#toc-intersections-of-object-types)

So with interfaces instead of types there would be a real difference to the original typings